### PR TITLE
Feat: add statisticsService

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Implement functionality to query for school averages](https://www.notion.so/uwblueprintexecs/Implement-functionality-to-query-for-school-averages-08591015e61f4156a54ebda26f28c009)

## Implementation description
* Created statisticsService - implementation and interface files
* Add StatisticsResponseDTO
* Add `getAverageScoresBySchool` with passing and failing tests

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/implementations/statisticsService.ts
Comment (carissa-tang) [inline backend/services/implementations/statisticsService.ts:94]: would prefer doing this directly in the aggregation pipeline but haven't found a good way to implement :(.  the closest solution would be using `$zip` but there's this [bug](https://jira.mongodb.org/browse/SERVER-27964?page=com.atlassian.streams.streams-jira-plugin%3Aactivity-stream-issue-tab) that restricts the function's use to arrays with fixed dimensions.

Comment (carissa-tang) [reply backend/services/implementations/statisticsService.ts:94]: essentially we want to transpose an array of test results (i.e. `[result1, result2, result3,...]` with each result being an array of student answers) to `[question1_answers, question2_answers, ...]` with each entry being an array of all answers for question `i`

**Comment by carissa-tang** [comment]
i should also refactor some of my code since i think some of it can be reused for the other pipelines - but will hold off until i get an idea of how the other ones will be implemented

File: backend/services/interfaces/statisticsService.ts
Comment (joyce-shi) [inline backend/services/interfaces/statisticsService.ts]: Should we name this something more specific? Since we will have this type of DTO for every category that we breakdown by (i.e school, sub-region, country)

File: backend/services/implementations/statisticsService.ts
Comment (joyce-shi) [inline backend/services/implementations/statisticsService.ts]: Not sure if I'm just brainfreezing, but why is this called `testSessionDtos`?

File: backend/services/implementations/statisticsService.ts
Comment (joyce-shi) [inline backend/services/implementations/statisticsService.ts]: Are we mapping boolean values -> 0/1 binary values here? If so, would it be more intuitive to leave this as `true/false` values here and then do a temporary conversion in `computeAverageScorePerQuestion`  instead? To avoid storing magic numbers.

File: backend/services/interfaces/statisticsService.ts
Comment (joyce-shi) [inline backend/services/interfaces/statisticsService.ts]: Do we need to store `TestResponseDTO` for each `SchoolResponseDTO`? 
i.e maybe it makes more sense to store `school: Array<SchoolResponseDTO>` instead of `SchoolResponseDTO`? Let me know your thoughts here!
maybe then, it would also eliminate the need for `mapStatisticsToStatisticsDTOs` helper?

**Review by joyce-shi — CHANGES_REQUESTED** [review]
Left a couple comments about about structure of the changes. Let me know if I'm missing the point of your PR, not sure if I'm completely understanding all aspects!

File: backend/services/implementations/statisticsService.ts
Comment (joyce-shi) [reply backend/services/implementations/statisticsService.ts:94]: I think it's okay to leave out of aggregation pipeline, as long as the meaning is clear! Which I think it is here.

**Comment by joyce-shi** [comment]
> Left a couple comments about about structure of the changes. Let me know if I'm missing the point of your PR, not sure if I'm completely understanding all aspects!

Tagging @mahadzaryab1 for a second pair of eyes.

File: backend/services/interfaces/statisticsService.ts
Comment (carissa-tang) [reply backend/services/interfaces/statisticsService.ts]: yes i was thinking that.. also wanted to see how others would implement (e.g. we could have a "filter/grouping" field instead that takes in either a school, sub-region or country). but might be more clear to keep them separate.

File: backend/services/implementations/statisticsService.ts
Comment (carissa-tang) [reply backend/services/implementations/statisticsService.ts]: LOL forgot to rename - thanks for catching that!

File: backend/services/interfaces/statisticsService.ts
Comment (carissa-tang) [reply backend/services/interfaces/statisticsService.ts]: so would we return `Array<StatisticsResponseDTO>` or `Array<SchoolResponseDTO>` ? i'm a bit confused by what you mean

File: backend/services/interfaces/statisticsService.ts
Comment (carissa-tang) [reply backend/services/interfaces/statisticsService.ts]: will make separate one for now! can refactor later if needed

Comment (carissa-tang) [reply backend/services/interfaces/statisticsService.ts]: actually, i'll refactor AggregateTestSessionDTO to be reusable and we can have separate response dtos

**Pull request closed without merging**
